### PR TITLE
feat: async stack trace middleware for compute-node gRPC

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This PR introduces a gRPC middleware for async stack trace. Now we can use `risectl trace` to check stack traces for both actors and gRPC servings.

```
--- gRPC Traces ---

>> gRPC 127.0.0.1:5689 (11438)
[captured 25.578916ms ago]
/stream_service.StreamService/BarrierComplete:11438 [112.661584ms]
  collect_barrier (epoch 2815842782150656) [112.611874ms]

>> gRPC 127.0.0.1:5689 (11440)
[captured 26.822375ms ago]
/stream_service.StreamService/BarrierComplete:11440 [101.70779ms]
  collect_barrier (epoch 2815842798927872) [101.648125ms]
```

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- #4533

### CODE REVIEW

**Comment by yuhao-su** [comment]
LGTM!

File: src/compute/src/server.rs [hunk 1]
```diff
@@ -182,6 +186,7 @@ pub async fn compute_node_serve(
         config.streaming.clone(),
     ));
     let source_mgr = Arc::new(MemSourceManager::new(source_metrics));
+    let grpc_stack_trace_mgr = GrpcStackTraceManagerRef::default();
```
Comment (yuhao-su) [inline src/compute/src/server.rs:189]: Maybe make the init style consistent with other mgrs?

**Review by skyzh — APPROVED** [review]
Better to have a doc on how to use this. Good work!

File: src/ctl/src/cmd_impl/trace.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    if all_actor_traces.is_empty() && all_grpc_traces.is_empty() {
+        println!(
+            "No traces found. No actors are running, or `{ASYNC_STACK_TRACE_KEY}` is not set?"
+        );
+    } else {
+        println!("--- Actor Traces ---");
+        for (key, trace) in all_actor_traces {
+            println!(">> Actor {key}\n{trace}");
+        }
+        println!("--- gRPC Traces ---");
```
Comment (fuyufjh) [inline src/ctl/src/cmd_impl/trace.rs:80]: BTW, "RPC trace" sounds more friendly because it doesn't matter which RPC framework is used internally?

**Review by fuyufjh — APPROVED** [review]
LGTM!

**Review by yuhao-su — APPROVED** [review]

**Pull request merged**